src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -826,6 +834,40 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType, bool inExpressionTr
                 refKind,
                 returnType);
 
+        private LambdaSymbol? _lazyLambdaSymbol = null;
+
+        internal LambdaSymbol? LazyInProgressLambdaSymbol
+        {
+            get
+            {
+                Interlocked.CompareExchange(ref _lazyLambdaSymbol, CreateTemporaryLambdaSymbol(), null);
```

Comment:
>CreateTemporaryLambdaSymbol() [](http://example.com/codeflow?start=67&length=29)

This is being called each time. Should the property check `if (_lazyLambdaSymbol is null)` first?